components/gitpod-protocol/go/gitpod-service.go

```diff
@@ -376,13 +408,19 @@ func (gp *APIoverJSONRPC) handler(ctx context.Context, conn *jsonrpc2.Conn, req
 
 	gp.mu.RLock()
 	defer gp.mu.RUnlock()
-	for chn := range gp.subs[instance.ID] {
+	for chn := range gp.instanceSubs[instance.ID] {
+		select {
+		case chn <- &instance:
+		default:
+		}
+	}
+	for chn := range gp.instanceSubs[""] {
```

Comment:
oh, it seems that you can also do already `conn.InstanceUpdates(ctx, "")` and then filter by workspace id on the client?